### DISCUSSION TITLE
how umami to calculate Visitors?

### DISCUSSION THREAD

**Discussion by caojinghuan** [umami-software/umami#2106] [Q&A]
I have a system got only 230 users in the database, but Umami Vistors count is 259. so my question is why the two number is different? how umami calculate Visitors number?

**Comment by briancao** [comment]
Umami calculates visitors using a hash of websiteId, hostname, ip, and userAgent.  The code can be seen here: https://github.com/umami-software/umami/blob/71bb012cabcff4d75bcf230f31be22d4eb6d10c6/lib/session.ts#L44

If a user changed devices, that would record under a different session.

**Comment by caojinghuan** [comment]
Got you, Thanks a lot Brian!
________________________________
发件人: Brian Cao ***@***.***>
发送时间: 2023年7月10日 4:55
收件人: umami-software/umami ***@***.***>
抄送: caojinghuan ***@***.***>; Author ***@***.***>
主题: Re: [umami-software/umami] how umami to calculate Visitors? (Discussion #2106)


Umami calculates visitors using a hash of websiteId, hostname, ip, and userAgent. The code can be seen here: https://github.com/umami-software/umami/blob/71bb012cabcff4d75bcf230f31be22d4eb6d10c6/lib/session.ts#L44

If a user changed devices, that would record under a different session.

―
